orc/src/main/java/org/apache/iceberg/orc/OrcSchemaVisitor.java

```diff
@@ -123,8 +122,8 @@
     return visitor.record(record, names, visitFields(fields, names, visitor));
   }
 
-  public String optionName() {
-    return "_option";
+  public String optionName(int ordinal) {
+    return "tag_" + ordinal;
```

Comment:
Could you give an example how this is used?

Reply:
It's used by the `ApplyNameMapping` visitor to correctly recognize the union converted struct field names, for example sth like `c1.tag_1.c2` and assign the correct ID to such field.

Reply:
I didn't check the ApplyNameMapping visitor, but can you verify that using same ordinal across unions - i.e., (1, "tag_0") and again (5, "tag_0") as in your test case -  could lead to any issues?

Reply:
It won't, the (1, tag_0) and (5, tag_0) are two fields at different nested levels inside the schema so there's no ambiguity.